src/controllers/user.controller.js

```diff
@@ -0,0 +1,93 @@
+'use strict';
+
+const userService = require('../service/user.service');
+
+const getAll = (req, res) => {
+  res.send(userService.get());
+};
+
+const getOne = (req, res) => {
+  const id = parseInt(req.params.id);
+
+  if (isNaN(id)) {
+    res.sendStatus(400);
+
+    return;
+  }
+
+  const user = userService.getById(id);
+
+  if (!user) {
+    res.sendStatus(404);
+
+    return;
+  }
+
+  res.send(user);
+};
+
+const addOne = (req, res) => {
+  const { name } = req.body;
+
+  if (typeof name !== 'string') {
+    res.sendStatus(400);
+
+    return;
+  }
+
+  const newUser = userService.create({ name });
+
+  res.statusCode = 201;
+  res.send(newUser);
+};
+
+const deleteOne = (req, res) => {
+  const id = parseInt(req.params.id);
+
+  if (isNaN(id)) {
+    res.sendStatus(400);
+
+    return;
+  }
+
+  if (!userService.getById(id)) {
+    res.sendStatus(404);
+
+    return;
+  }
+
+  userService.remove(id);
+  res.sendStatus(204);
+};
+
+const updateOne = (req, res) => {
+  const id = parseInt(req.params.id);
+  const { name } = req.body;
+
+  if (isNaN(id) || typeof name !== 'string') {
+    res.sendStatus(400);
```

Comment:
if `typeof name !== 'string'` better to send 422

Reply:
`typeof name !== 'string'` checks for correct data type as well as for name not being undefined (being provided by client) and in this case, according to task requirements I should send 400. Should I distinguish validation for existence and validation for correct data type and send 400 and 422 respectively in all responses?